horovod/runner/elastic/discovery.py

```diff
@@ -39,13 +52,48 @@ def get_event(self):
     def set_event(self):
         self._event.set()
 
+    def _in_cooldown_period(self, current_time):
+        return self._cooldown_period_end_ts > current_time
+
+
+    def _set_cooldown_period(self, current_time):
+        if self._cooldown_lower_limit == -1 or self._cooldown_upper_limit == -1:
+            return
+        self._blacklist_count += 1
+        def _exponential_backoff_time():
+            cooldown_delay = self._cooldown_lower_limit * (1 << self._blacklist_count)
+            + (random.uniform(0,1) * self._cooldown_lower_limit)
+            logging.debug(f"{self._blacklist_count}:{self._cooldown_period_end_ts} cooldown_delay: {cooldown_delay}")
+            return max(self._cooldown_lower_limit, min(self._cooldown_upper_limit, cooldown_delay))
```

Comment:
What is the edge case that requires the `max`here?

Reply:
I added it for symmetry with the `min` computation, but I don't see the need unless we support fractional cooldown limits.

Reply:
well, since you never assert that `cooldown_lower_limit` is smaller than `cooldown_upper_limit`, the `max` is needed here